### ISSUE TITLE
Accessible links with bigger clickable areas

### ISSUE THREAD

**Issue by mayank99** [iTwin/iTwinUI#840]
Many of our components are often used with links. Lets look at two specific examples: a `MenuItem` and a `Tile`.

In the case of MenuItem, there is a built-in padding. When adding an anchor element inside it, it makes the clickable area smaller and also brings the default anchor styles.
```html
<li class="iui-menu-item">
  <a href='#'>...</a>
</li>
```
![](https://user-images.githubusercontent.com/9084735/206545436-2a425c7e-a91c-4583-9960-a22fe06ba519.png)

To work around this, users often add `<a>` around the `<li>` but that is no longer valid markup (`<li>` needs to be a direct child of `<ul>`/`<ol>`). It also adds multiple tab stops. 👎

In the case of Tile, users often want the whole tile to be clickable to they add `<a>` around the tile, which also breaks accessibility because screen readers often only read out the first element inside links. 👎

The accessible markup would look more like:
```html
<div class="iui-tile">
  <div class="iui-tile-thumbnail">...</div>
  <div class="iui-tile-content">
    <div class="iui-tile-name">
      <a href='#'>The title of the card</a>
    </div>
    <div class="iui-tile-description">The stuff outside the link</div>
  </div>
</div>
```

This is achievable by using a pseudo-element that increases the clickable portion to be as big as the wrapping component. 👍
```scss
.iui-tile, .iui-menu-item {
  position: relative;

  a::before {
    content: '';
    position: absolute;
    inset: 0;
  }
}
```

Note that this approach should be used with caution and there should be no other interactive elements inside the wrapping component. In the case of menu item we would also need to fix multiple tab stops (in JS).

Further reading: https://inclusive-components.design/cards/

**Comment by gretanausedaite** [comment]
Tile portion was done with #959

**Comment by mayank99** [comment]
List was done with #1106
